confgenerator/resourcedetector/gce_detector.go

```diff
@@ -128,6 +133,50 @@ func (r GCEResource) MonitoredResource() *monitoredres.MonitoredResource {
 	}
 }
 
+func (r GCEResource) PrometheusStyleMetadata() map[string]string {
```

Comment:
I think this leaks abstraction — before, the translation from resources to Prometheus labels was completely defined in `prometheus.go`, and now it's in all of the resource definitions. It might even be better to instead have a type switch in `prometheus.go` that knows about all the resources.

That said, I'll defer to @quentinmit for the final opinion.

Reply:
I think it's better to keep it in resource detector; that way there's only one file that needs to be changed (or duplicated) to support new resources in the future.